File: Java/PrisonerSimulator/src/Player/Strategy/EvilRepeaterStrategy.java
package Player.Strategy;

import Game.Decision;
import Player.IPlayerStrategyFacade;

import java.util.List;
import java.util.Optional;

public class EvilRepeaterStrategy implements IStrategy {
    @Override
    public Decision MakeDecision(IPlayerStrategyFacade player, int rivalId) {
        var rivalDecisions = player.getRivalDecisionsById(rivalId);
        if (Optional.ofNullable(rivalDecisions).map(List::isEmpty).orElse(true))
            return Decision.Defect;

        return rivalDecisions.get(rivalDecisions.size() - 1);
    }
}

File: Kotlin/PrisonerSimulator/src/player/strategy/EvilRepeaterStrategy.kt
package player.strategy

import game.Decision
import player.IPlayerStrategyFacade

class EvilRepeaterStrategy : IStrategy {
    override fun makeDecision(player: IPlayerStrategyFacade, rivalId: Int): Decision =
        player.rivalDecisionsById[rivalId]?.lastOrNull() ?: Decision.Defect
}
